WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -431,6 +454,28 @@ private extension DashboardViewModel {
         }
     }
 
+    @MainActor
+    func updateHasOrdersStatus() async {
+        hasOrders = await loadHasOrdersStatus()
+    }
+
+    @MainActor
+    func loadHasOrdersStatus() async -> Bool {
+        await withCheckedContinuation { continuation in
+            stores.dispatch(OrderAction.checkIfStoreHasOrders(siteID: self.siteID, onCompletion: { [weak self] result in
+                guard let self else { return }
+
+                switch result {
+                case .success(let hasOrders):
+                    continuation.resume(returning: hasOrders)
+                case .failure(let error):
+                    DDLogError("⛔️ Dashboard (Share Your Store) — Error checking if site has orders: \(error)")
+                    continuation.resume(returning: self.hasOrders) // Use original value on error
```

Comment:
How about rethrowing the error, catching it in the callsite, and then doing nothing if an error occurs? IMO that would simplify the logic and avoid capturing `self` here.

Reply:
Handled in https://github.com/woocommerce/woocommerce-ios/pull/12521/commits/22a586c520aeb8592ecd7d66d91841b4c7349ea0